plugins/Hosting/js/sagas/saga.js

```diff
@@ -204,6 +206,33 @@ function *requestDefaultSettingsSaga() {
 	}
 }
 
+// getHostStatus returns a promise that resolves with `true` if the provided
+// netaddress is connectable, or `false` if it is not connectable. The input
+// 'hostAddr' must be an address in the URL format, eg:
+// tcp://123.456.789.0:9982
+const getHostStatus = (hostAddr) => new Promise((resolve) => {
+	const hostUrl = url.parse(hostAddr)
+	const socket = net.connect(hostUrl.port, hostUrl.hostname)
+	socket.setTimeout(15e3) // 15 second host connectability timeout
```

Comment:
That's probably too abstract. If it's not immediately obvious what the thing means, we're going to have 100 users asking us what it means and why it's not 100%, etc.

I think what would be best would be to have a single field in the bar, like we already have, that only displays if you are actively hosting (or have contracts), which either says 'Host is Working' (or something to that effect), 'Host cannot be Reached' (or something to that effect), or 'Host is not Visible to Renters' (or something to that effect). And if we add more potential ways to detect that something in the host does not seem to be working, we'll add those too to the potential failure messages.

And if multiple failures are active, we'll have a sensible priority. For example, if your ports aren't forwarded, then it doesn't really make sense to point out that renter's can't contact you. Just mentioning the port thing should be enough.